allennlp/nlp_api.py

```diff
@@ -0,0 +1,261 @@
+       `sequence` of vectors, while ``get_sentence_encoder`` returns a `single` vector.  Applicable
+       modules here are CNNs, tree-structured recursive networks, RNNs where you just take the last
+       hidden state or do some pooling operation over the sequence, etc.
+
+    All of these functions behave similarly: you give the function a name and a fallback behavior.
+    The name should correspond to a location in your model, such as "passage_tokens",
+    "question_encoder", "modeling_layer_2", etc.  Each time you call these APIs in different parts
+    of your model code, you should typically pass a unique name to the function.  When constructing
+    this ``NlpApi`` object, you map those names to concrete ``Modules`` that will be used to
+    perform the function.  The ``fallback_behavior`` parameter makes it so you can be less verbose
+    in constructing that mapping, allowing you to have multiple unique names in your model
+    correspond to the same key in the mapping.  This parameter determines what to do when a name
+    you requested (like "modeling_layer_2") is not in the mapping you provided.  There are three
+    options:
+
+        - ``"crash"``: raise an error.  This is typically the default.  The intention is to help
+          you find bugs - if you specify a particular name when constructing your model without
+          giving a fallback behavior, you must also provide a mapping for that name in the
+          ``NlpApi``.
+        - ``"new default"``: In this case, we return a new module created with the ``default_*``
+          function passed to the constructor.  If this is your fallback behavior, you will get a
+          `new` module by default when you give a new name.  You should use this for things like
+          stacked context encoders, or for the first time you get a sentence encoder you want to be
+          shared (see example below).
+        - ``"use [name]``: In this case, we `reuse` the module with the given name, essentially
+          replacing the ``name`` given as an argument to this function with the name in the
+          fallback behavior.  You should use this fallback behavior for modules that should be
+          shared between input fields by default, e.g., for encoding a passage and a question using
+          the same RNN, or for embedding several different TextFields (again, see example below).
+
+    For example, if we're writing model code for the Bidirectional Attention Flow model (BiDAF) for
+    reading comprehension on the Stanford Question Answering Dataset (SQuAD), we might have code
+    that looks something like this::
+
+        def forward(self, question, passage, answer_span_begin=None, answer_span_end=None):
+            # This is our first token embedder, so we get a new default.
+            question_embedder = nlp_api.get_token_embedder("question",
+                                                           fallback_behavior="new default")
```

Comment:
For the suggestion part, consider using standard language idioms instead of rolling your own. The less the library requires the user to remember, the easier is to onboard and retain casual users.